themes/default/content/blog/serverless-architecture-design/index.md

```diff
@@ -0,0 +1,116 @@
+
+    ``` go
+    // The 'date' function that returns the current date
+	mux.HandleFunc("/api/date", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+		w.Header().Set("Access-Control-Allow-Methods", "GET")
+
+		if r.Method == http.MethodOptions {
+			w.WriteHeader(http.StatusNoContent)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.Write([]byte(fmt.Sprintf(`{ "now": %d }`, time.Now().UnixNano()/1000000)))
+	})
+
+    // The 'weather' function that returns the weather
+	mux.HandleFunc("/api/weather", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+		w.Header().Set("Access-Control-Allow-Methods", "GET")
+
+		if r.Method == http.MethodOptions {
+			w.WriteHeader(http.StatusNoContent)
+			return
+		}
+
+		weather := getWeatherData()
+
+		// Convert the weather data to JSON.
+		responseJSON, err := json.Marshal(weather)
+		if err != nil {
+			http.Error(w, "Error encoding response to JSON", http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write(responseJSON)
+	})
+    ```
+
+- **Statelessness**: Serverless functions should be stateless. This encourages scalability and fault tolerance since every serverless function call should be able to function on its own, regardless of previous or subsequent calls. The recommended practise for managing data that has to be persistent beyond the span of one function execution is to store it externally in reliable data storage systems like object stores or databases. In doing so, serverless applications maintain the clear separation between function logic and stateful data management while guaranteeing data durability, consistency, and adaptability to changing workloads. The stateless nature of HTTP is an important aspect that makes serverless computing a good fit for many HTTP-based use cases. In a stateless protocol like HTTP, each request from a client to a server is treated independently, with no connection or context maintained between requests. This aligns well with the design philosophy of serverless computing, where each function execution is isolated and stateless Serverless platforms are designed to handle short-lived, stateless functions that respond to individual HTTP requests, making them well-suited for web applications, microservices, and API endpoints.
+
+> Figure 1. Resources needed in a serverless architecture
+![serverless-resources](./serverless-resources.png)
+
+- **Function triggers**: When dealing with serverless functions you should choose the Right trigger that best suits your use case. Common triggers include HTTP requests, message queues, database changes, and timers. In our application, we used an HTTP trigger for our two functions.
```

Comment:
```suggestion
- **Choose the right trigger**: When dealing with serverless functions you should choose the right trigger that best suits your use case. Common triggers include HTTP requests, message queues, database changes, and timers. In our application, we used an HTTP trigger for our two functions.
```

Reply:
I don't get this bullet. Its not so much choosing the right trigger but choosing the right manner to transfer data. The data source determines the event which determines the trigger.

Reply:
You are basically saying you should choose event-driven versus synchronous / poll-based architectures?

Maybe you are saying choose event-driven or poll-based based on your use case right? If that is the case, perhaps you need to update the text a bit

Reply:
> If you're building something that is dependent on DB changes, one might think it makes sense to make a http triggered function and check if the db has changed at the start of the function, but a cosmosdb triggered function (speaking in the context if Azure) makes more sense.
**This** is the sort of content/recommendation that should be in the article, IMO.